homeassistant/components/media_player/kodi.py

```diff
@@ -130,6 +130,8 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
         password=config.get(CONF_PASSWORD),
         turn_off_action=config.get(CONF_TURN_OFF_ACTION), websocket=websocket)
 
+    key = 'media_player.' + slugify(config.get(CONF_NAME))
+    hass.data[DATA_KODI][key] = entity
     async_add_devices([entity], update_before_add=True)
```

Comment:
Somehow this was missed before, but this is a coroutine. If we make this `yield from async_add_devices(...`, the `entity_id` should be available afterwards. Use `entity.entity_id` as the dict key.